src/olympia/reviewers/management/commands/notify_about_auto_approve_delay.py

```diff
@@ -17,21 +18,36 @@ def fetch_versions_waiting_for_approval_for_too_long(self):
         """
         Return versions that are auto-approvable but have waited long enough
         that we need to notify their developers about the delay.
+
+        Versions are grouped by add-on, only the latest auto-approvable version
+        for each add-on is considered ; if a version has been created recently
+        enough for a given add-on and not auto-approved, none of its other
+        versions will be returned.
         """
-        waiting_period = datetime.now() - timedelta(
-            hours=self.WAITING_PERIOD_HOURS)
         exclude_kwargs = {
             'addon__addonreviewerflags__notified_about_auto_approval_delay':
                 True
         }
-        return (
+        qs = (
+            # Base queryset with auto-approvable versions. Default ordering
+            # is reset to make the GROUP BY work.
             Version.objects
                    .auto_approvable()
-                   .filter(created__lt=waiting_period)
                    .exclude(**exclude_kwargs)
-                   .order_by('created')
-                   .distinct()
+                   .order_by()
         )
+        # Get only the latest version for each add-on from the auto-approvable
+        # versions.
+        latest_per_addon = qs.values('addon').annotate(latest_pk=Max('pk'))
```

Comment:
This doesn't take into account channels - there could be an upload as a listed version and an unlisted version.  As the unlisted version doesn't supersede the listed version (or visa versa) should emails be sent for both?

Reply:
My idea was to avoid sending multiple emails, regardless of what the versions are.

Reply:
I guess that fine - as long as there's nothing channel specific in the email sent.